### PR TITLE
Nevis system Frame Rollover fix

### PR DESCRIPTION
**Pull request by ibsafa**
### Description

Introduces two fixes: 
1) Correction to the GPS timestamp issued by NTB. When nevis clock frames roll over in between a PPS pulse and a trigger, the assigned trigger timestamp is ambiguous for one second. this fix ensures that 1 second of data is timestamped correctly. 
2) NTB and TPC boardreaders introduce a correction to the frame number (adding multiples of 2^24 when to account for rollovers)

### Testing details
- *Run number associated with test*: 
- Other information:

### CODE REVIEW

**Comment by ibsafa** [comment]
We have determined that we don't need PR#113 in sbndaq-artdaq-core, and just board reader changes are enough to introduce the fixes we want. I will commit to this PR one more time and then it will be ready for review. will update with a comment here when that's done.

**Comment by ibsafa** [comment]
Made changes to remove artdaq-core changes. @afilkins @eyandel This PR is now ready for review. I believe test runs are planned for today. Thank you!

**Comment by gennadiy-fnal** [comment]
@ibsafa  There are merge conflicts that need to be addressed.

**Comment by ibsafa** [comment]
@gennadiy-fnal fixed it, thanks!

File: sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc [hunk 1]
```diff
@@ -252,16 +257,53 @@ bool sbndaq::NevisTB_generatorBase::FillNTBFragment(artdaq::FragmentPtrs &frags,
...
+    TLOG(TLVL_INFO) << "Frame difference (between GPS and trigger): " << frame_diff;
+
+    if(tframe < prevFrame){
+      //If rollover happens, current frame number will be smaller than previous frame number. in that case, increment the rollover counter
+      rollCounter+=1;
+      TLOG(TLVL_INFO) << "Trigger frames rolled over!!!! " << " this many times: " << rollCounter;
+    }
+
+    corrFrame = tframe + rollCounter*16777216;
+    TLOG(TLVL_INFO) << " Corrected Frame:  " << corrFrame <<  " Uncorrected Frame: " << tframe;
```
Comment (afilkins) [inline sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc]: Can we change this to Debug? it spams the log messages pretty heavily

File: sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc [hunk 2]
```diff
@@ -252,16 +257,53 @@ bool sbndaq::NevisTB_generatorBase::FillNTBFragment(artdaq::FragmentPtrs &frags,
...
+    clock_gettime(CLOCK_REALTIME, &unixtime);
+
+    //When the nevis clock rolls over (frames go back to 0) in between a PPS and a trigger
+    //catch the rollover by calculating the difference between PPS tagged time and trigger tagged time
+    int32_t   frame_diff = abs(ntbheader->getFrame() - GPSframe);
+    //long long tolerance  = 16777216 - FramesPerSecond_ - 10; //16777216 = 2^24 is when the rollover happens. frame numbers are stored in 24 bits
+    int32_t   tolerance  = 16770000;
+    int32_t   corrFrame;
+
+    TLOG(TLVL_INFO) << "Frame difference (between GPS and trigger): " << frame_diff;
```
Comment (afilkins) [inline sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc]: DEBUG for this one too?

File: sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc (hunk 1 above)
Comment (ibsafa) [reply sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc]: would I still be able to see it in the log files?

Comment (afilkins) [reply sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc]: It should, but another option would be to make it a TRACE instead which would show up in logs but not the message viewer

**Review by ibsafa — COMMENTED** [review]
@afilkins I changed the traces to DEBUG as requested. thanks for looking at this!

File: sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc (hunk 2 above)
Comment (ibsafa) [reply sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc]: done. for this one I just commented out. it's not needed anymore

File: sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc (hunk 1 above)
Comment (ibsafa) [reply sbndaq-artdaq/Generators/SBND/NevisTPC/NevisTB_generatorBase.cc]: done

**Review by afilkins — APPROVED** [review]

**Pull request merged**